### PR TITLE
Mviews never sync after migrations

### PR DESCRIPTION
**Pull request by evtechltd**
The mview sync on every migration is pretty painful - I'd rather make it manual for now and figure something out later

### CODE REVIEW

**Review by amackie-zyper — COMMENTED** [review]
Dug into this a little, I'm not sure we can completely rely on the scheduled rebuilds to completely maintain the mviews. It might work this time but cause future problems?

All `refresh()` does is call `REFRESH MATERIALIZED VIEW` in postgres https://github.com/zyperco/django-pgviews/blob/master/django_pgviews/view.py#L268 which has problems when the underlying tables/sql change https://stackoverflow.com/a/47686154

That looks like the reasoning mentioned in https://github.com/mypebble/django-pgviews/issues/38 which is why the `ViewSyncer` forces an update to drop and rebuild on each migrate https://github.com/zyperco/django-pgviews/blob/master/django_pgviews/apps.py#L35 because it doesn't examine the migrations in detail. 

We might need to change our code to understand more knowledge about what mviews need to get updated, but thats a lot of work.

**Comment by evtechltd** [comment]
yeah, that's right, but it's no better really to be relying on migrations to rebuild the mviews. obviously the best thing would be to rebuild them based on the content of the changes, but that's a ton of work; the next best thing is to rebuild them regularly, which rebuilding on migration doesn't achieve. rebuilding on migration also makes the build take an hour and a half, which also means the previous build is talking to a migrated database for the length of time the mviews take to build which can cause a bunch of errors.

tl;dr this isn't the right way to rebuild the mviews, but neither is the unchanged version - i want to push this so i can promote and then we can figure out how to rebuild the mviews properly later

**Comment by evtechltd** [comment]
@amackie-zyper

**Comment by evtechltd** [comment]
also worth saying that the refresh behaviour for the old rails app was the same ie. we're not losing anything by going back to it

**Review by amackie-zyper — APPROVED** [review]
Okay this makes sense as a quick push to speed up deployment.

I think post-migration is the only sane place to rebuild a view properly, but we need to do something about the times sooner rather than later. Rebuilding periodically completely after the migration has happened and the app is deployed is also wrong in a different way

**Comment by evtechltd** [comment]
cc @rhngit @Qu4tro

**Comment by rhngit** [comment]
Makes sense for now. NB Rails only rebuilds the mview in the migration, so is slightly better here. We’ll move away from the big mviews soon. Generally I’d advocate replacing most of them as a 1h refresh rate has some problems (and results in some temporary inconsistencies - campaign stats for one).
